lib/pharos/phases/validate_host.rb

```diff
@@ -89,15 +92,18 @@ def cpu_arch
       # @return [Hash]
       def host_checks
         data = {}
-        result = @ssh.exec("sudo curl -sSf --connect-timeout 1 --cacert /etc/kubernetes/pki/ca.crt https://localhost:6443/healthz")
-        data['api_healthy'] = (result.success? && result.stdout == 'ok')
-        data['ca_exists'] = @ssh.file('/etc/kubernetes/pki/ca.key').exist?
         data['kubelet_configured'] = @ssh.file('/etc/kubernetes/kubelet.conf').exist?
+        data['ca_exists'] = @ssh.file('/etc/kubernetes/pki/ca.key').exist?
+        data['etcd_ca_exists'] = @ssh.file('/etc/pharos/pki/ca-key.pem').exist?
+
+        if data['ca_exists']
+          result = @ssh.exec("sudo curl -sSf --connect-timeout 1 --cacert /etc/kubernetes/pki/ca.crt https://localhost:6443/healthz")
```

Comment:
Unrelated change, but we don't need to bother with the `curl` if the CA cert that it's using doesn't exist?

Reply:
True.